examples/hello.rs

```diff
@@ -5,29 +5,28 @@ extern crate pretty_env_logger;
 extern crate tokio;
 
 use futures::Future;
-use futures::future::lazy;
 
 use hyper::{Body, Response};
-use hyper::server::{Http, const_service, service_fn};
+use hyper::server::{Server, const_service, service_fn};
 
 static PHRASE: &'static [u8] = b"Hello World!";
 
 fn main() {
     pretty_env_logger::init();
+
     let addr = ([127, 0, 0, 1], 3000).into();
 
     let new_service = const_service(service_fn(|_| {
         //TODO: when `!` is stable, replace error type
         Ok::<_, hyper::Error>(Response::new(Body::from(PHRASE)))
     }));
 
-    tokio::run(lazy(move || {
-        let server = Http::new()
-            .sleep_on_errors(true)
-            .bind(&addr, new_service)
-            .unwrap();
+    let server = Server::bind(&addr)
+        .expect("bind to address")
```

Comment:
I'm strongly considering making the default `Server::bind` either a) panic if binding errors, or b) lazily binding on the first `poll` of the `Server`. Forcing users to deal with an error that isn't likely to happen every single time feels like an annoying speed bump. Anyone wanting to handle a bind error can bind a `TcpListener` themselves, and use `Builder::new`.